guides/rails_guides/markdown.rb

```diff
@@ -109,7 +109,7 @@ def generate_structure
                   hierarchy = hierarchy[0, 3] + [node]
                 end
 
-                node[:id] = dom_id(hierarchy) unless node[:id]
+                node[:id] = dom_id(hierarchy).parameterize unless node[:id]
```

Comment:
Should not `dom_id` take care of this?

Reply:
It is! By using percent encode! But it's kinda an ugly solution on some urls its just hard and long to read the text. But i understand that this it's just my opinion.

Reply:
This change will also probably break some anchors that are already linked in the internet. Not that we should care, but if `dom_id` is already taking care of escaping special characters why should we change it?

Reply:
Yeah, on second thought doesn't seem fair to break something over what it is perfect fine already. Thanks for the insights!